lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -3490,6 +3518,26 @@ static bool allowsApproxFunction(IntrinsicInst *II) {
            cast<VectorType>(Ty)->getElementType()->isFloatTy()));
 }
 
+bool allowDecorateWithBufferLocationINTEL(IntrinsicInst *II) {
+  SmallVector<Value *, 8> UserList;
+
+  for (auto *Inst : II->users()) {
```

Comment:
Should we use [stripPointerCasts](https://llvm.org/doxygen/classllvm_1_1Value.html#a966eb231e7d4e572874d2cb49b18faea) here?

Reply:
Probably do not need to. The reason is the same as above.